tools/loader/ImageClassifier.cpp

```diff
@@ -321,26 +349,47 @@ int main(int argc, char **argv) {
     parseInputImageList(inputImageListFile);
   }
 
+  // Stream input mode.
   const bool streamInputFilenamesMode =
       inputImageFilenames.size() == 1 && inputImageFilenames.front() == "-";
 
   GLOW_ASSERT(!(streamInputFilenamesMode && emittingBundle()) &&
               "Cannot emit a bundle and also stream inputs.");
 
+  // Mini-batch mode.
+  const bool miniBatchMode = miniBatch > 0;
+  GLOW_ASSERT(((!miniBatchMode) || (!streamInputFilenamesMode)) &&
+              "The minibatch option is not compatible with the stream input "
+              "image mode.");
+  GLOW_ASSERT(
+      ((!miniBatchMode) || (inputImageFilenames.size() % miniBatch == 0)) &&
+      "The number of input images must be a multiple of the mini-batch.");
+
   // Used to make sure we only compile once, and run only once if not streaming.
   bool isFirstRun = true;
 
   // These will be set during the first run.
   Placeholder *inputImagePH = nullptr;
   Tensor *SMT = nullptr;
-
+  size_t minibatchIndex = 0;
   Tensor inputImageData;
+  std::vector<std::string> inputImageBatchFilenames;
```

Comment:
nit: Realizing now that we could just re-use `inputImageFilenames` instead of adding this new `inputImageBatchFilenames`. We were already re-using `inputImageFilenames` inside of `getNextImageFilenames()`, which is pretty similar to `getNextMiniBatch()`. Not a huge deal, but would remove an extra `std::vector<std::string>`.

Reply:
We actually need both `inputImageBatchFilenames` and `inputImageFilenames`for mini(batching). The first contains the full list of images while the second contains only images of the current mini-batch. Its different from getting the image list from stdin because we have the full list of images at the beginning with mini-batching. 
`getNextMiniBatch(inputImageBatchFilenames, inputImageFilenames, minibatchIndex, miniBatch)`

Reply:
Ah right, got it!